From 234201104b9bf5b1fcc622a607cc2618143ac4ac Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Fri, 12 Sep 2014 14:35:34 -0400
Subject: [tools] perf: Allow to specify lib compile variable for spec usage

Message-id: <1410532537-3812-2-git-send-email-jolsa@redhat.com>
Patchwork-id: 94042
O-Subject: [PATCH RHEL7.1 BZ879133 1/4] perf tools: Allow to specify lib compile variable for spec usage
Bugzilla: 879133
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Petr Holasek <pholasek@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 879133
https://bugzilla.redhat.com/show_bug.cgi?id=879133

upstream
========
commit 941446d841cfbb4103834885189bf8973007bce7
Author: Jiri Olsa <jolsa@kernel.org>
Date:   Mon Aug 25 16:55:52 2014 +0200

description
===========
We need a way to specify $(lib) part of the installation
path for traceevent plugin libraries. Currently we use
'lib64' for x86_64 and 'lib' otherwise.

Instead of listing all possible values, this change allows
the rpm spec code to specify the correct $(lib) part based
on processed architecture, like

  $ make ... lib=%{_lib}

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index f30ac5e..568f22a 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -621,11 +621,13 @@ else
 sysconfdir = $(prefix)/etc
 ETC_PERFCONFIG = etc/perfconfig
 endif
+ifndef lib
 ifeq ($(IS_X86_64),1)
 lib = lib64
 else
 lib = lib
 endif
+endif # lib
 libdir = $(prefix)/$(lib)
 
 # Shell quote (do not use $(call) to accommodate ancient setups);
-- 
1.7.1